<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>400</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="1">
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>原始数据处理</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="0">
         <widget class="QGroupBox" name="groupBox_3">
          <property name="title">
           <string>标注软件</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_4">
           <item row="0" column="0">
            <widget class="QLabel" name="label_7">
             <property name="text">
              <string>软件路径</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QPushButton" name="pushButton_path_labelimg">
             <property name="text">
              <string>选择路径</string>
             </property>
            </widget>
           </item>
           <item row="0" column="2">
            <widget class="QLineEdit" name="lineEdit_path_soft"/>
           </item>
           <item row="0" column="3">
            <widget class="QPushButton" name="pushButton_run_labelimg">
             <property name="text">
              <string>启动</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QGroupBox" name="groupBox_2">
          <property name="title">
           <string>训练集和测试集随机挑选程序</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_3">
           <item row="0" column="0">
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>图像文件夹</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QPushButton" name="pushButton_path_choice_img">
             <property name="text">
              <string>选择路径</string>
             </property>
            </widget>
           </item>
           <item row="0" column="2" colspan="2">
            <widget class="QLineEdit" name="lineEdit_path_choice_img"/>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>标签文件夹</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QPushButton" name="pushButton_path_choice_label">
             <property name="text">
              <string>选择路径</string>
             </property>
            </widget>
           </item>
           <item row="1" column="2" colspan="2">
            <widget class="QLineEdit" name="lineEdit_path_choice_label"/>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string>dataset路径</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QPushButton" name="pushButton_path_dataset">
             <property name="text">
              <string>选择路径</string>
             </property>
            </widget>
           </item>
           <item row="2" column="2" colspan="2">
            <widget class="QLineEdit" name="lineEdit_path_dataset"/>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="label_6">
             <property name="toolTip">
              <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;选择测试集占测试集+训练集的比例，建议选择10%-15%。&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
             </property>
             <property name="whatsThis">
              <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
             </property>
             <property name="text">
              <string>测试集比例</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QSpinBox" name="spinBox_ratio_test">
             <property name="suffix">
              <string>%</string>
             </property>
             <property name="prefix">
              <string/>
             </property>
             <property name="maximum">
              <number>30</number>
             </property>
             <property name="value">
              <number>10</number>
             </property>
            </widget>
           </item>
           <item row="3" column="2">
            <widget class="QCheckBox" name="checkBox_auto_delete">
             <property name="toolTip">
              <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;勾选该选项，则会先删除dataset文件夹下的所有文件，然后重新创建image、label文件夹及其子文件夹train和val。&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
             </property>
             <property name="text">
              <string>自动删除</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="3" column="3">
            <widget class="QPushButton" name="pushButton_run_pick">
             <property name="text">
              <string>运行</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QGroupBox" name="groupBox">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;将命名不规范的图像和标签文件规范化命名，会自动对应图像和标签文件，只有图像或者标签文件也能运行。&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="title">
           <string>重命名工具</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_5">
           <item row="0" column="0">
            <widget class="QLabel" name="label">
             <property name="text">
              <string>图像文件夹</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QPushButton" name="pushButton_path_rename_img">
             <property name="text">
              <string>选择路径</string>
             </property>
            </widget>
           </item>
           <item row="0" column="2">
            <widget class="QLineEdit" name="lineEdit_path_rename_img"/>
           </item>
           <item row="0" column="3" rowspan="2">
            <widget class="QPushButton" name="pushButton_run_rename">
             <property name="text">
              <string>运行</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>标签文件夹</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QPushButton" name="pushButton_path_rename_label">
             <property name="text">
              <string>选择路径</string>
             </property>
            </widget>
           </item>
           <item row="1" column="2">
            <widget class="QLineEdit" name="lineEdit_path_rename_label"/>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>训练</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_8">
        <item row="0" column="0">
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>模型地址</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0" colspan="3">
         <widget class="QGroupBox" name="groupBox_6">
          <property name="title">
           <string>训练参数</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_6">
           <item row="0" column="0">
            <widget class="QRadioButton" name="radioButton_train_restart">
             <property name="text">
              <string>重新训练</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
             <attribute name="buttonGroup">
              <string notr="true">buttonGroup</string>
             </attribute>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QCheckBox" name="checkBox_train_pretrained">
             <property name="text">
              <string>权值预载</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="0" column="2">
            <widget class="QLabel" name="label_9">
             <property name="text">
              <string>训练轮数</string>
             </property>
            </widget>
           </item>
           <item row="0" column="3">
            <widget class="QSpinBox" name="spinBox_val_epoch">
             <property name="maximum">
              <number>10000</number>
             </property>
             <property name="value">
              <number>300</number>
             </property>
            </widget>
           </item>
           <item row="1" column="0" colspan="2">
            <widget class="QRadioButton" name="radioButton_train_resume">
             <property name="text">
              <string>继续训练</string>
             </property>
             <property name="checked">
              <bool>false</bool>
             </property>
             <attribute name="buttonGroup">
              <string notr="true">buttonGroup</string>
             </attribute>
            </widget>
           </item>
           <item row="1" column="3">
            <widget class="QPushButton" name="pushButton_run_train">
             <property name="text">
              <string>开始训练</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_11">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;选择yaml文件&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="text">
           <string>数据地址</string>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QLineEdit" name="lineEdit_path_yaml"/>
        </item>
        <item row="3" column="0" colspan="3">
         <widget class="QGroupBox" name="groupBox_4">
          <property name="title">
           <string>训练结果</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_7">
           <item row="0" column="0" colspan="4">
            <widget class="QProgressBar" name="progressBar_train">
             <property name="value">
              <number>0</number>
             </property>
            </widget>
           </item>
           <item row="1" column="0" colspan="4">
            <widget class="QTextBrowser" name="textBrowser_train"/>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_10">
             <property name="text">
              <string>已用时间:</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QLabel" name="label_train_time">
             <property name="text">
              <string>----</string>
             </property>
            </widget>
           </item>
           <item row="2" column="2">
            <widget class="QLabel" name="label_12">
             <property name="text">
              <string>预估剩余时间:</string>
             </property>
            </widget>
           </item>
           <item row="2" column="3">
            <widget class="QLabel" name="label_trian_resum_time">
             <property name="text">
              <string>----</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QLineEdit" name="lineEdit_path_model"/>
        </item>
        <item row="0" column="1">
         <widget class="QPushButton" name="pushButton_choice_model">
          <property name="text">
           <string>选择文件</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QPushButton" name="pushButton_choice_yaml">
          <property name="text">
           <string>选择文件</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_3">
       <attribute name="title">
        <string>测试</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_9">
        <item row="0" column="0">
         <widget class="QLabel" name="label_16">
          <property name="text">
           <string>测试模型路径</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QPushButton" name="pushButton_path_check_model">
          <property name="text">
           <string>选择路径</string>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QLineEdit" name="lineEdit_path_check_model"/>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_14">
          <property name="text">
           <string>测试素材路径</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QPushButton" name="pushButton_path_check">
          <property name="text">
           <string>选择路径</string>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QLineEdit" name="lineEdit_path_check"/>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_15">
          <property name="text">
           <string>素材保存路径</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QPushButton" name="pushButton_path_save">
          <property name="text">
           <string>选择路径</string>
          </property>
         </widget>
        </item>
        <item row="2" column="2">
         <widget class="QLineEdit" name="lineEdit_path_save"/>
        </item>
        <item row="3" column="0">
         <widget class="QPushButton" name="pushButton_run_check">
          <property name="text">
           <string>运行</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>600</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="buttonGroup"/>
 </buttongroups>
</ui>
